return_type.md

```diff
@@ -0,0 +1,94 @@
+    return_type(f, T::Type{<:Tuple}) -> R::Type
+
+Return a type `R` such that `f(args...) isa R` where `args isa T`.
+
+The type `R` obtained from `return_type` is merely an upper bound.  There may
+exist a stricter type `S` such that `f(args...) isa S` for every `args isa T`
+with `S <: R` and `S != R`. Furthermore, the exact type `R` obtained from
+`return_type` depends on various factors including but not limited to the exact
+Julia version used, packages loaded, and command line options. As such, when
+used in publicly registered packages, **it is the package authors'
+responsibility to ensure that the API guarantees provided by the package do not
+depend on the exact type `R` obtained from `return_type`.**
+
+# Extended help
+
+## Examples
+
+The following function is an invalid use-case of `return_type`.
```

Comment:
Great example!

Reply:
Thanks! These are based on my pet peeve and I guess that's why it's realistic :)